File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.cs
public class Solution {
    public bool AreOccurrencesEqual(string s) {
        var dic = new Dictionary<int, int>();
        
        foreach(char c in s){
            if(dic.ContainsKey(c)) dic[c]++;
            else dic[c] = 1;
        }
        
        var piv = dic[s[0]];
        
        return dic.Where(w=>w.Value != piv).Count() == 0;
    }
}

File: 1941-check-if-all-characters-have-equal-number-of-occurrences/1941-check-if-all-characters-have-equal-number-of-occurrences.ts
function areOccurrencesEqual(s: string): boolean {
    const map = new Map<string, number>();
    
    for(let c of s){
        map.set(c, (map.get(c) || 0) + 1);
    }
    const arr = new Set(map.values());
    return arr.size === 1;
};